modin/experimental/core/execution/native/implementations/hdk_on_native/dataframe/dataframe.py

```diff
@@ -2508,10 +2508,13 @@ def from_arrow(cls, at, index_cols=None, index=None, columns=None):
             new_columns = pd.Index(data=at.column_names, dtype="O")
             new_index = pd.RangeIndex(at.num_rows)
 
-        new_dtypes = pd.Series(
-            [cls._arrow_type_to_dtype(col.type) for col in at.columns],
-            index=at.column_names,
-        )
+        try:
+            new_dtypes = pd.Series(
+                [cls._arrow_type_to_dtype(col.type) for col in at.columns],
+                index=at.column_names,
+            )
+        except NotImplementedError:
+            new_dtypes = at.to_pandas().dtypes
```

Comment:
can we convert to pandas only a subset of "bad" columns and not the whole arrow table?